backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarWriteFilesExec.scala

```diff
@@ -58,81 +96,77 @@ class VeloxColumnarWriteFilesExec(
-          val jsonObject = objectMapper.readTree(fragments.toString)
-
-          val fileWriteInfos = jsonObject.get("fileWriteInfos").elements()
-          if (jsonObject.get("fileWriteInfos").elements().hasNext) {
-            val writeInfo = fileWriteInfos.next();
-            numBytes += writeInfo.get("fileSize").longValue()
-            // Get partition information.
-            if (jsonObject.get("name").textValue().nonEmpty) {
-              val targetFileName = writeInfo.get("targetFileName").textValue()
-              val partitionDir = jsonObject.get("name").textValue()
-              updatedPartitions += partitionDir
-              val tmpOutputPath =
-                writeFilesSpec.description.path + "/" + partitionDir + "/" + targetFileName
-              val absOutputPathObject =
-                writeFilesSpec.description.customPartitionLocations.get(
-                  PartitioningUtils.parsePathFragment(partitionDir))
-              if (absOutputPathObject.nonEmpty) {
-                val absOutputPath = absOutputPathObject.get + "/" + targetFileName
-                addedAbsPathFiles(tmpOutputPath) = absOutputPath
+          objectMapper.registerModule(DefaultScalaModule)
+          for (i <- 0 until loadedCb.numRows() - 1) {
+            val fragments = loadedCb.column(1).getUTF8String(i + 1)
+            val metrics = objectMapper
+              .readValue(fragments.toString.getBytes("UTF-8"), classOf[VeloxWriteFilesMetrics])
+            logDebug(s"Velox write files metrics: $metrics")
+
+            val fileWriteInfos = metrics.fileWriteInfos
+            assert(fileWriteInfos.length == 1)
+            val fileWriteInfo = fileWriteInfos.head
+            numBytes += fileWriteInfo.fileSize
+            val targetFileName = fileWriteInfo.targetFileName
+            val outputPath = writeFilesSpec.description.path
+
+            // part1=1/part2=1
+            val partitionFragment = metrics.name
+            // write a non-partitioned table
+            if (partitionFragment != "") {
+              updatedPartitions += partitionFragment
+              val tmpOutputPath = outputPath + "/" + partitionFragment + "/" + targetFileName
+              val customOutputPath = writeFilesSpec.description.customPartitionLocations.get(
+                PartitioningUtils.parsePathFragment(partitionFragment))
+              if (customOutputPath.isDefined) {
+                addedAbsPathFiles(tmpOutputPath) = customOutputPath.get + "/" + targetFileName
               }
             }
           }
-        }
 
-        // TODO: need to get the partition Internal row?
-        val stats = BasicWriteTaskStats(Seq.empty, loadedCb.numRows() - 1, numBytes, numWrittenRows)
-        val summary =
-          ExecutedWriteSummary(updatedPartitions = updatedPartitions, stats = Seq(stats))
+          // Reports bytesWritten and recordsWritten to the Spark output metrics.
+          Option(TaskContext.get()).map(_.taskMetrics().outputMetrics).foreach {
```

Comment:
Why need to update the task metrics?

Reply:
it is used to show the task/stage metrics in UI, e.g.,
<img width="1585" alt="image" src="https://github.com/oap-project/gluten/assets/12025282/5e549c76-75c6-4604-b6f2-bca587aaef36">